File: problems/3501-3600/3508_implement-router/starter.go
package main

type Router struct{}

func NewRouterTyped(memoryLimit int) *Router {
	panic("TODO")
}

func (design *Router) addPacket(source int, destination int, timestamp int) bool {
	panic("TODO")
}

func (design *Router) forwardPacket() []int {
	panic("TODO")
}

func (design *Router) getCount(destination int, startTime int, endTime int) int {
	panic("TODO")
}

File: problems/3501-3600/3508_implement-router/starter.rs
pub struct Router;

impl Router {
    pub fn new(memoryLimit: i32) -> Self {
        panic!("TODO")
    }

    pub fn addPacket(&mut self, source: i32, destination: i32, timestamp: i32) -> bool {
        panic!("TODO")
    }

    pub fn forwardPacket(&mut self) -> Vec<i32> {
        panic!("TODO")
    }

    pub fn getCount(&mut self, destination: i32, startTime: i32, endTime: i32) -> i32 {
        panic!("TODO")
    }
}
